spec/models/family/celebrity_family_spec.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Family
+  RSpec.describe CelebrityFamily, type: :model, versioning: true do
+    describe "#reify" do
+      context "belongs_to" do
+        it "uses the correct item_type in queries" do
+          parent = described_class.new(name: "Jermaine Jackson")
+          parent.path_to_stardom = "Emulating Motown greats such as the Temptations and "\
+                                   "The Supremes"
```

Comment:
😀

Reply:
Ah yes -- I have a distinct interest in making a quasi-boring topic (auditing) into something enduring.

And actually I would hope that things like email notifications and cool reporting with pretty charts could piggy-back on top of what we store in `versions`!

Reply:
> ...do we need to provide a convenient way for people convert their existing database records?

We could, and in our project I actually have a migration that does this.  It's possible to include with this a generator that creates a migration to cycle through existing PaperTrail::Version entries and update item_type appropriately.

-= But =-  wonderfully enough this is not entirely essential because ActiveRecord just does the right thing; reified objects get instantiated as their proper subclassed type.  With the Person / Doctor / Patient example, if for a Doctor you have a combo of entries from the past (<= v9.1) where the item_type was stored as Person, and now there are other entries (>= v9.2) some changes stored as Doctor, the reification always brings back a Doctor.

The AT part currently does not rehydrate all the historic attributes properly when this update is applied, so that's the reason for the couple notes in `spec/models/pet_spec.rb`. During the coming week I have high hopes to solve that and then have the best of all worlds.  The biggest "win" in my mind becomes more specific reporting of the item_type, and certainly having the example in `spec/models/person.rb` work out is a wonderful benefit as well.